### PR TITLE
Updated Status Bar Color and Created a Separate  Theme File

### PR DESCRIPTION
**Pull request by 2002Bishwajeet**
This PR fixes Status bar color which got merged with AppBar color and the icons were not visible clearly
Image 1: Not able to see icons clearly
image
Image 2: The fix done
Screenshot_20210510-204247_ruTorrent_Mobile

Also Created a separate theme file and transferred all the themedata details to it (organizing the code)

### CODE REVIEW

**Review by pratikbaid3 — CHANGES_REQUESTED** [review]

File: lib/components/data_input.dart [hunk 1]
```diff
@@ -48,7 +48,12 @@ class DataInput extends StatelessWidget {
             border: InputBorder.none,
             contentPadding: EdgeInsets.symmetric(horizontal: 16, vertical: 16),
             hintText: hintText,
-            hintStyle: TextStyle(color: hintTextColor ?? borderColor),
+            hintStyle: TextStyle(
+                color: hintTextColor == null
```
[the remark this answers is not in the document]
Comment (2002Bishwajeet) [reply lib/components/data_input.dart]: Done 👍

**Review by tusharojha — APPROVED** [review]
LGTM!

**Comment by pratikbaid3** [comment]
@2002Bishwajeet on running the changes you made, this is what I get
Simulator Screen Shot - iPhone 12 Pro Max - 2021-05-13 at 23 51 38

```
======== Exception caught by widgets library =======================================================
The following _TypeError was thrown building DataInput(dirty):
type 'Color' is not a subtype of type 'bool'

The relevant error-causing widget was: 
  DataInput file:///Users/pratikbaid/Desktop/Work/FlutterDev/rutorrent-flutter/lib/screens/configurations_screen.dart:162:25
When the exception was thrown, this was the stack: 
#0      DataInput.build (package:rutorrentflutter/components/data_input.dart:52:38)
#1      StatelessElement.build (package:flutter/src/widgets/framework.dart:4569:28)
#2      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4495:15)
#3      Element.rebuild (package:flutter/src/widgets/framework.dart:4189:5)
#4      ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:4474:5)
...
====================================================================================================
```

File: lib/components/data_input.dart [hunk 2]
```diff
@@ -48,7 +48,10 @@ class DataInput extends StatelessWidget {
             border: InputBorder.none,
             contentPadding: EdgeInsets.symmetric(horizontal: 16, vertical: 16),
             hintText: hintText,
-            hintStyle: TextStyle(color: hintTextColor ?? borderColor),
+            hintStyle: TextStyle(
+                color: hintTextColor ?? Provider.of<Mode>(context).isLightMode
```
Comment (pratikbaid3) [inline lib/components/data_input.dart:52]: This is the cause of the issue below. What you are trying to do is evaluate a variable of type Color as a bool. Using '?.' might resolve this issue.
Simulator Screen Shot - iPhone 12 Pro Max - 2021-05-13 at 23 51 38

**Review by 2002Bishwajeet — COMMENTED** [review]
Ok Reverted to previous commit

**Review by pratikbaid3 — APPROVED** [review]

**Comment by pratikbaid3** [comment]
There are some issues in the changes you made that are causing the screen to now have a grey color instead of the original white color when the theme was set to light.
Simulator Screen Shot - iPhone 12 Pro Max - 2021-05-14 at 00 29 31

Simulator Screen Shot - iPhone 12 Pro Max - 2021-05-14 at 00 29 22

**Comment by 2002Bishwajeet** [comment]
That's because I have set the brightness to dark  . To fix this I would have to specify background color to every widget.

Can you guide me how to get past the login screen?

**Comment by pratikbaid3** [comment]
@2002Bishwajeet several issues still exist. Unfortunately, the only way to resolve them is to get a ruTorrent web running on your local machine. Please refer to the readme and the slack thread for any help

File: lib/utilities/themes.dart [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      dialogBackgroundColor: Colors.white,
+      cardColor: Colors.white);
+  static ThemeData darkTheme(BuildContext context) => ThemeData(
+        scaffoldBackgroundColor: kBackgroundDT,
+        canvasColor: kBackgroundDT,
+        textTheme: ThemeData.dark().textTheme.apply(
+              fontFamily: 'SFUIDisplay',
+            ),
+        brightness: Brightness.dark,
+        backgroundColor: Colors.black,
```
Comment (rj-since-2000) [inline lib/utilities/themes.dart:24]: The background for dark theme is too dark/black. Change it back to kBackgroundDT.

File: lib/pages/rss_feeds.dart [hunk 4]
```diff
@@ -27,8 +28,14 @@ class _RSSFeedsState extends State<RSSFeeds> {
...
       return Scaffold(
+        backgroundColor: Theme.of(context).backgroundColor,
         body: RefreshIndicator(
-          color: Theme.of(context).primaryColorLight,
+          backgroundColor: Provider.of<Mode>(context).isLightMode
+              ? Colors.white
+              : Colors.black,
+          color: Provider.of<Mode>(context).isLightMode
+              ? Colors.black
+              : Colors.white,
```
Comment (rj-since-2000) [inline lib/pages/rss_feeds.dart:38]: Don't use hard coded colours like black and white, instead use the constants defined in utilities.

**Review by rj-since-2000 — CHANGES_REQUESTED** [review]
Make the necessary changes and squash your commits with appropriate messages.

**Review by rj-since-2000 — APPROVED** [review]

**Comment by rj-since-2000** [comment]
@2002Bishwajeet fix format exceptions

**Comment by 2002Bishwajeet** [comment]
@rj-since-2000  Done 👍

**Pull request merged**